jwql/instrument_monitors/nirspec_monitors/ta_monitors/wata_monitor.py

```diff
@@ -770,6 +766,32 @@ def update_ta_success_txtfile(self):
                 line = "{:<50} {:<50}".format(suc, ta_failure[idx])
                 txt.write(line + "\n")
 
+    def read_existing_html(self):
+        """
+        This function gets the data from the Bokeh html file created with
+        the NIRSpec TA monitor script.
+        """
+        self.output_dir = os.path.join(get_config()['outputs'], 'wata_monitor')
+        ensure_dir_exists(self.output_dir)
+
+        self.output_file_name = os.path.join(self.output_dir, "wata_layout.html")
+        if not os.path.isfile(self.output_file_name):
+            return 'No WATA data available', '', ''
+
+        # open the html file and get the contents
+        with open(self.output_file_name, "r") as html_file:
+            contents = html_file.read()
+
+        soup = BeautifulSoup(contents, 'html.parser').body
+
+        # find the script elements
+        script1 = str(soup.find('script', type='text/javascript'))
+        script2 = str(soup.find('script', type='application/json'))
```

Comment:
I wrote it this way because of a difference between the way the components are returned from the bokeh `components` call and the way they are written to an html file with the `save` call.  The first one returns one script, which combines the JSON data and the javascript components, the second splits them into two scripts (one javascript, one JSON).  It seems to be identical in functionality, as long as both scripts from `save` appear wherever the single script from `components` would.

Reply:
Thank you for the explanation, I didn't know this was possible with bokeh, it's really cool. This is really useful for future monitors/plotting purposes.